src/Akka.Persistence.Sql.Linq2Db/Query/Dao/BaseByteReadArrayJournalDAO.cs

```diff
@@ -141,95 +140,35 @@ private static async Task AddTagDataFromTagTable(List<JournalRow> toAdd, DataCon
                                     .Take(input.maxTake).ToListAsync();
                             })
                         .Via(_deserializeFlow);
+                
                 case TagReadMode.TagTable:
-                    return EventByTagTableOnly(tag, offset, maxOffset, separator, maxTake);
+                    return AsyncSource<JournalRow>.FromEnumerable(
+                            new { separator, tag, offset, maxOffset, maxTake, ConnectionFactory },
+                            async input =>
+                            {
+                                await using var conn = input.ConnectionFactory.GetConnection();
+                                var journalTable = conn.GetTable<JournalRow>();
+                                var tagTable = conn.GetTable<JournalTagRow>();
+                                var query =
+                                    from r in journalTable
+                                    from lp in tagTable.Where(jtr => 
+                                        jtr.OrderingId == r.Ordering 
+                                        && jtr.PersistenceId == r.PersistenceId).DefaultIfEmpty()
```

Comment:
Why do we need to join on `PeristenceId` here? Is this _just_ for deletes?

Reply:
You're right, let me fix that

Reply:
@Arkatufus we should probably have a unit test that would have caught this on the read-side - is there one sitting in an unmerged PR somewhere?

Reply:
Not yet, unit testing will be another PR after this